### ISSUE TITLE
Please upgrade lodash to 4.17.10

### ISSUE THREAD

**Issue by Innei** [BryanDonovan/node-cache-manager#141]
https://github.com/Automattic/mongoose/issues/6507

_Originally posted by @fbudassi in https://github.com/Automattic/mongoose/issues/6507#issuecomment-392583454_

**Comment by Innei** [comment]
Below version 4.17.10 will cause mongoose error 
```
Unhandled rejection TypeError: Class constructor CoreMongooseArray cannot be invoked without 'new'
```
please

**Comment by Innei** [comment]
please remove lodash.clonedeep and replace with loadsh. lodash.clonedeep package is too old.

**Comment by BryanDonovan** [comment]
Hi.  I don't understand why the `lodash.clonedeep` version in `cache-manager` affects mongoose..

**Comment by Innei** [comment]
I don't understand too, but I update lodash version instead of lodash.clonedeep, it works
https://github.com/Automattic/mongoose/issues/6507#issuecomment-392435501

**Comment by BryanDonovan** [comment]
Are you saying that you have a project that uses both `cache-manager` and `mongoose`, and you're not using the `memory` store from `cache-manager`, and when you update `cache-manager` to use lodash 4.17.0, then the error goes away?

**Comment by Innei** [comment]
I didnot set store, and I viewed code sources, the default store is `memory`. so I using the `memory` store

https://github.com/BryanDonovan/node-cache-manager/blob/227fb2119e5e124ab9e6895beac13cbe11bccbf5/lib/caching.js#L27

**Comment by Innei** [comment]
https://github.com/mx-space/server/blob/master/libs/common/src/common.module.ts

**Comment by BryanDonovan** [comment]
Can you please tell me what the output of `npm ll lodash` is in your project?

**Comment by Innei** [comment]
ok,

```
  │ git+https://github.com/PicGo/PicGo-Core.git
  │ https://github.com/PicGo/PicGo-Core#readme
  ├─┬ inquirer@6.5.2
  │ │ A collection of common interactive command line user interfaces.
  │ │ git+https://github.com/SBoudrias/Inquirer.js.git
  │ │ https://github.com/SBoudrias/Inquirer.js#readme
  │ └── lodash@4.17.15  deduped
  │     Lodash modular utilities.
  │     git+https://github.com/lodash/lodash.git
  │     https://lodash.com/
  ├─┬ lowdb@1.0.0
  │ │ Small JSON database for Node, Electron and the browser. Powered by Lodash.
  │ │ git://github.com/typicode/lowdb.git
  │ │ https://github.com/typicode/lowdb
  │ └── lodash@4.17.15  deduped
  │     Lodash modular utilities.
  │     git+https://github.com/lodash/lodash.git
  │     https://lodash.com/
  └─┬ request-promise-native@1.0.8
    │ The simplified HTTP request client 'request' with Promise support. Powered by native ES6 promises
.
    │ git+https://github.com/request/request-promise-native.git
    │ https://github.com/request/request-promise-native#readme
    └─┬ request-promise-core@1.1.3
      │ Core Promise support implementation for the simplified HTTP request client 'request'.
      │ git+https://github.com/request/promise-core.git
      │ https://github.com/request/promise-core#readme
      └── lodash@4.17.15  deduped
          Lodash modular utilities.
          git+https://github.com/lodash/lodash.git
          https://lodash.com/
```

**Comment by BryanDonovan** [comment]
Thanks, just wanted to rule out the possibility of a different version of lodash being installed somewhere.  I'll try to get this updated by tomorrow.

**Comment by Innei** [comment]
```
➜  server git:(master) npm ll lodash
server@0.2.0
│ /root/server
│ Mix Space Server
├─┬ @nestjs/cli@7.1.4
│ │ Nest - modern, fast, powerful node.js web framework (@cli)
│ │ git+https://github.com/nestjs/nest-cli.git
│ │ https://github.com/nestjs/nest-cli#readme
│ └─┬ inquirer@7.1.0
│   │ A collection of common interactive command line user interfaces.
│   │ git+https://github.com/SBoudrias/Inquirer.js.git
│   │ https://github.com/SBoudrias/Inquirer.js#readme
│   └── lodash@4.17.15  deduped
│       Lodash modular utilities.
│       git+https://github.com/lodash/lodash.git
│       https://lodash.com/
├─┬ @nestjs/swagger@4.5.3
│ │ Nest - modern, fast, powerful node.js web framework (@swagger)
│ │ git+https://github.com/nestjs/swagger.git
│ │ https://github.com/nestjs/swagger#readme
│ └── lodash@4.17.15
│     Lodash modular utilities.
│     git+https://github.com/lodash/lodash.git
│     https://lodash.com/
├─┬ @typegoose/auto-increment@0.4.1
│ │ Automaticly Increment properties
│ │ git+https://github.com/typegoose/auto-increment.git
│ │ https://github.com/typegoose/auto-increment
│ └── lodash@4.17.15  deduped
│     Lodash modular utilities.
│     git+https://github.com/lodash/lodash.git
│     https://lodash.com/
├─┬ @typegoose/typegoose@6.5.0
│ │ Define Mongoose models using TypeScript classes
│ │ git+https://github.com/typegoose/typegoose.git
│ │ https://typegoose.github.io/typegoose/
│ └── lodash@4.17.15  deduped
│     Lodash modular utilities.
│     git+https://github.com/lodash/lodash.git
│     https://lodash.com/
├─┬ @typescript-eslint/parser@2.29.0
│ │ An ESLint custom parser which leverages TypeScript ESTree
│ │ git+https://github.com/typescript-eslint/typescript-eslint.git
│ │ https://github.com/typescript-eslint/typescript-eslint#readme
│ └─┬ @typescript-eslint/typescript-estree@2.29.0
│   │ A parser that converts TypeScript source code into an ESTree compatible form
│   │ git+https://github.com/typescript-eslint/typescript-eslint.git
│   │ https://github.com/typescript-eslint/typescript-eslint#readme
│   └── lodash@4.17.15  deduped
│       Lodash modular utilities.
│       git+https://github.com/lodash/lodash.git
│       https://lodash.com/
├─┬ cache-manager@3.2.1
│ │ Cache module for Node.js
│ │ git+https://github.com/BryanDonovan/node-cache-manager.git
│ │ https://github.com/BryanDonovan/node-cache-manager#readme
│ └── lodash@4.17.15
│     Lodash modular utilities.
│     git+https://github.com/lodash/lodash.git
│     https://lodash.com/
├─┬ eslint@6.8.0
│ │ An AST-based pattern checker for JavaScript.
│ │ git+https://github.com/eslint/eslint.git
│ │ https://eslint.org
│ ├── lodash@4.17.15  deduped
│ │   Lodash modular utilities.
│ │   git+https://github.com/lodash/lodash.git
│ │   https://lodash.com/
│ └─┬ table@5.4.6
│   │ Formats data into a string table.
│   │ git+https://github.com/gajus/table.git
│   │ https://github.com/gajus/table#readme
│   └── lodash@4.17.15  deduped
│       Lodash modular utilities.
│       git+https://github.com/lodash/lodash.git
│       https://lodash.com/
├─┬ jest@25.4.0
│ │ Delightful JavaScript Testing.
│ │ git+https://github.com/facebook/jest.git
│ │ https://jestjs.io/
│ └─┬ @jest/core@25.4.0
│   │ Delightful JavaScript Testing.
│   │ git+https://github.com/facebook/jest.git
│   │ https://jestjs.io/
│   ├─┬ @jest/reporters@25.4.0
│   │ │ Jest's reporters
│   │ │ git+https://github.com/facebook/jest.git
│   │ │ https://jestjs.io/
│   │ └─┬ istanbul-lib-instrument@4.0.1
│   │   │ Core istanbul API for JS code coverage
│   │   │ git+ssh://git@github.com/istanbuljs/istanbuljs.git
│   │   │ https://istanbul.js.org/
│   │   └─┬ @babel/traverse@7.9.5
│   │     │ The Babel Traverse module maintains the overall tree state, and is responsible for replacing, removing, and adding nodes
│   │     │ https://github.com/babel/babel/tree/master/packages/babel-traverse
│   │     │ https://babeljs.io/
│   │     └── lodash@4.17.15  deduped
│   │         Lodash modular utilities.
│   │         git+https://github.com/lodash/lodash.git
│   │         https://lodash.com/
│   ├─┬ @jest/transform@25.4.0
│   │ │ git+https://github.com/facebook/jest.git
│   │ │ https://github.com/facebook/jest#readme
│   │ └─┬ @babel/core@7.9.0
│   │   │ Babel compiler core.
│   │   │ https://github.com/babel/babel/tree/master/packages/babel-core
│   │   │ https://babeljs.io/
│   │   ├─┬ @babel/generator@7.9.5
│   │   │ │ Turns an AST into code.
│   │   │ │ https://github.com/babel/babel/tree/master/packages/babel-generator
│   │   │ │ https://babeljs.io/
│   │   │ └── lodash@4.17.15  deduped
│   │   │     Lodash modular utilities.
│   │   │     git+https://github.com/lodash/lodash.git
│   │   │     https://lodash.com/
│   │   ├─┬ @babel/helper-module-transforms@7.9.0
│   │   │ │ Babel helper functions for implementing ES6 module transformations
│   │   │ │ https://github.com/babel/babel/tree/master/packages/babel-helper-module-transforms
│   │   │ │ https://babeljs.io/
│   │   │ └── lodash@4.17.15  deduped
│   │   │     Lodash modular utilities.
│   │   │     git+https://github.com/lodash/lodash.git
│   │   │     https://lodash.com/
│   │   └── lodash@4.17.15  deduped
│   │       Lodash modular utilities.
│   │       git+https://github.com/lodash/lodash.git
│   │       https://lodash.com/
│   └─┬ jest-snapshot@25.4.0
│     │ git+https://github.com/facebook/jest.git
│     │ https://github.com/facebook/jest#readme
│     └─┬ @babel/types@7.9.5
│       │ Babel Types is a Lodash-esque utility library for AST nodes
│       │ https://github.com/babel/babel/tree/master/packages/babel-types
│       │ https://babeljs.io/
│       └── lodash@4.17.15  deduped
│           Lodash modular utilities.
│           git+https://github.com/lodash/lodash.git
│           https://lodash.com/
└─┬ picgo@1.4.8
  │ A tool for picture uploading
  │ git+https://github.com/PicGo/PicGo-Core.git
  │ https://github.com/PicGo/PicGo-Core#readme
  ├─┬ inquirer@6.5.2
  │ │ A collection of common interactive command line user interfaces.
  │ │ git+https://github.com/SBoudrias/Inquirer.js.git
  │ │ https://github.com/SBoudrias/Inquirer.js#readme
  │ └── lodash@4.17.15  deduped
  │     Lodash modular utilities.
  │     git+https://github.com/lodash/lodash.git
  │     https://lodash.com/
  ├─┬ lowdb@1.0.0
  │ │ Small JSON database for Node, Electron and the browser. Powered by Lodash.
  │ │ git://github.com/typicode/lowdb.git
  │ │ https://github.com/typicode/lowdb
  │ └── lodash@4.17.15  deduped
  │     Lodash modular utilities.
  │     git+https://github.com/lodash/lodash.git
  │     https://lodash.com/
  └─┬ request-promise-native@1.0.8
    │ The simplified HTTP request client 'request' with Promise support. Powered by native ES6 promises.
    │ git+https://github.com/request/request-promise-native.git
    │ https://github.com/request/request-promise-native#readme
    └─┬ request-promise-core@1.1.3
      │ Core Promise support implementation for the simplified HTTP request client 'request'.
      │ git+https://github.com/request/promise-core.git
      │ https://github.com/request/promise-core#readme
      └── lodash@4.17.15  deduped
          Lodash modular utilities.
          git+https://github.com/lodash/lodash.git
          https://lodash.com/
```

**Comment by Innei** [comment]
> Thanks, just wanted to rule out the possibility of a different version of lodash being installed somewhere. I'll try to get this updated by tomorrow.

please check the latest one, the top one is incomplete.

**Comment by Innei** [comment]
and I just change to the latest `lodash` version in `memory.js`

![image](https://user-images.githubusercontent.com/41265413/80779194-7195f200-8b9d-11ea-95d8-32abcd8b5797.png)

**Comment by BryanDonovan** [comment]
Cool, that makes sense.

**Comment by BryanDonovan** [comment]
This is fixed in version 3.3.0, available now.  Thanks for the bug report!

**Comment by Innei** [comment]
Thank you for your reply.
